SLAM/iterative_closest_point/iterative_closest_point.py

```diff
@@ -48,15 +49,20 @@ def icp_matching(previous_points, current_points):
         indexes, error = nearest_neighbor_association(previous_points, current_points)
         Rt, Tt = svd_motion_estimation(previous_points[:, indexes], current_points)
 
+        dError = preError - error
+        print("Residual:", error)
+
+        if dError < 0:  # prevent matrix H changing, exit loop
+            print("Not Converge...", preError, dError, count)
+            break
+
+        preError = error
+
         # update current points
         current_points = (Rt @ current_points) + Tt[:, np.newaxis]
```

Comment:
Could you please move line 61 and 62 to line 50 after svd_motion_estimation? I think this looks more easy to read the code.